apps/meteor/tests/end-to-end/api/24-methods.js

```diff
@@ -123,6 +126,276 @@ describe('Meteor.methods', function () {
+
+						const receiptsUserIds = [data.result[0].userId, data.result[1].userId];
+						expect(receiptsUserIds).to.have.members([credentials['X-User-Id'], user._id]);
+					});
+			});
+
+			it("should return only the sender's read receipt for a message sent in a thread after the main room is read", async () => {
+				await request
+					.post(methodCall('getReadReceipts'))
+					.set(credentials)
+					.send({
+						message: JSON.stringify({
+							method: 'getReadReceipts',
+							params: [{ messageId: firstThreadMessage._id }],
+							id: 'id',
+							msg: 'method',
+						}),
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.a.property('success', true);
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(1);
+						expect(data.result[0]).to.have.property('userId', credentials['X-User-Id']);
+					});
+			});
+
+			it('should read thread messages with the invited user', async () => {
+				await request
+					.post(methodCall('getThreadMessages'))
+					.set(userCredentials)
+					.send({
+						message: JSON.stringify({
+							id: 'id',
+							msg: 'method',
+							method: 'getThreadMessages',
+							params: [
+								{
+									tmid: firstMessage._id,
+								},
+							],
+						}),
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.a.property('success', true);
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(2);
+					});
+			});
+
+			it("should return the invited user's read receipt for a message sent in a thread after it is read", async () => {
+				await request
+					.post(methodCall('getReadReceipts'))
+					.set(credentials)
+					.send({
+						message: JSON.stringify({
+							method: 'getReadReceipts',
+							params: [{ messageId: firstThreadMessage._id }],
+							id: 'id',
+							msg: 'method',
+						}),
+					})
+					.expect('Content-Type', 'application/json')
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.a.property('success', true);
+						expect(res.body).to.have.a.property('message').that.is.a('string');
+
+						const data = JSON.parse(res.body.message);
+						expect(data).to.have.a.property('result').that.is.an('array');
+						expect(data.result.length).to.equal(2);
+
+						const receiptsUserIds = [data.result[0].userId, data.result[1].userId];
+						expect(receiptsUserIds).to.have.members([credentials['X-User-Id'], user._id]);
+					});
+			});
+
+			it('should mark the thread as read', async () => {
```

Comment:
This one is also data manipulation, that is not even being validated by anything.